File: elixir/lib/binary_search/binary_search_test.ex
ExUnit.start()

defmodule BinarySearchTest do
  use ExUnit.Case

  test "finds the item at the start of the list" do
    assert BinarySearch.search([1, 2, 3, 8, 9], 1) == {:ok, 0}
  end

  test "finds the item near the start of the list" do
    assert BinarySearch.search([1, 2, 3, 8, 9], 2) == {:ok, 1}
  end

  test "finds the item near in the middle of the list" do
    assert BinarySearch.search([20, 30, 40, 60, 75, 88], 40) == {:ok, 2}
  end

  test "finds the item at the end of the list" do
    assert BinarySearch.search([23, 48, 88, 97, 98, 99], 99) == {:ok, 5}
  end

  test "finds the correct number" do
    assert BinarySearch.search(
             [2738, 2889, 3004, 3483, 8484, 9999, 10005, 98373, 888_888, 982_687, 999_992],
             982_687
           ) == {:ok, 9}
  end
end

File: javascript/binarySearch/binarySearch.test.js
const binarySearch = require("./binarySearch.js");

test("Not finding the item should be -1", () => {
  expect(binarySearch([1, 2, 3, 4, 5, 6], 1)).toBe(0);
});

test("Finds the item at the start of the list", () => {
  expect(binarySearch([1, 2, 3, 4, 5, 6], 1)).toBe(0);
});

test("Finds the item at the middle of the list", () => {
  expect(binarySearch([11, 23, 25, 38, 49], 25)).toBe(2);
});

test("Finds the item at the end of the list", () => {
  expect(binarySearch([88, 891, 998, 999, 1003, 5999, 9999, 10002], 10002)).toBe(7);
});
